### PR TITLE
Add tests and docstrings to functions

### PR DESCRIPTION
**Pull request by davidstrouk**

### CODE REVIEW

File: gamla/data.py [hunk 1]
```diff
@@ -51,32 +54,39 @@ def _freeze_nonterminal(v):
...
 def tuple_of_tuples_to_csv(
     tuple_of_tuples: Tuple[Tuple[Any], ...],
     separator: Text = "\t",
 ) -> Text:
+    """
+    Return a CSV file string given a tuple of tuples. Each element is separated by the character "separator" (default is \t).
+
+    >>> tuple_of_tuples_to_csv((("name", "age"), ("David", "23"), ("Itay", "26")))
+    "name\tage\nDavid\t23\nItay\t26"
+    >>> tuple_of_tuples_to_csv((("name", "age"), ("David", "23"), ("Itay", "26")), "")
```
Comment (erez-work) [inline gamla/data.py:87]: I think you meant `" "` (not `""`)

File: gamla/functional.py [hunk 2]
```diff
@@ -259,10 +273,23 @@ def invoke(x):
 
 @currying.curry
 def assoc_in(d, keys, value, factory=dict):
+    """
+    Associate a new value to an existing dict given the path "keys".
```
Comment (erez-work) [inline gamla/functional.py:277]: I think this function returns a new dict and not changing the given one (I'm not 100% sure)

File: gamla/functional.py [hunk 3]
```diff
@@ -259,10 +273,23 @@ def invoke(x):
...
+
+    >>> assoc_in({"a": {"b": 1}}, ["a", "b"], 2)
+    {'a': {'b': 2}}
+    """
     return update_in(d, keys, lambda x: value, value, factory)
 
 
 def add_key_value(key, value):
+    """
+    Associate a new key-value pair to an existing dict.
```
Comment (erez-work) [inline gamla/functional.py:287]: Same here

**Review by erez-work — APPROVED** [review]

File: gamla/data.py [hunk 4]
```diff
@@ -51,32 +54,39 @@ def _freeze_nonterminal(v):
...
+#: >>> freeze_deep({"1": {"2": "345", "some-string": ["hello"]}})
+#: data.frozendict(
+#:     {"1": data.frozendict({"2": "345", "some-string": ("hello",)})},
+#: )
 freeze_deep = functional_generic.map_dict(_freeze_nonterminal, functional.identity)
 
 
 def csv_to_json(csv_file_path) -> List:
+    """
+    Return a JSON object given a CSV file path.
```
[the remark this answers is not in the document]
Comment (davidstrouk) [reply gamla/data.py:68]: Renamed to `csv_to_list_of_dicts`

**Pull request merged**